cincinnati/src/plugins/internal/arch_filter.rs

```diff
@@ -0,0 +1,271 @@
+//! be encoded as part of the _build_ information according to the SemVer specification.
+use crate::plugins::{
+    AsyncIO, BoxedPlugin, InternalIO, InternalPlugin, InternalPluginWrapper, PluginSettings,
+};
+use commons::GraphError;
+use failure::Fallible;
+use futures::Future;
+use prometheus::Registry;
+
+static DEFAULT_KEY_FILTER: &str = "io.openshift.upgrades.graph";
+static DEFAULT_ARCH_KEY: &str = "release.arch";
+
+#[derive(Clone, Debug, Deserialize, SmartDefault)]
+#[serde(default)]
+pub struct ArchFilterPlugin {
+    #[default(DEFAULT_KEY_FILTER.to_string())]
+    pub key_prefix: String,
+
+    #[default(DEFAULT_ARCH_KEY.to_string())]
+    pub key_suffix: String,
+}
+
+impl PluginSettings for ArchFilterPlugin {
+    fn build_plugin(&self, _: Option<&Registry>) -> Fallible<BoxedPlugin> {
+        Ok(new_plugin!(InternalPluginWrapper(self.clone())))
+    }
+}
+
+impl ArchFilterPlugin {
+    pub const PLUGIN_NAME: &'static str = "arch-filter";
+
+    /// Validate plugin configuration and fill in defaults.
+    pub fn deserialize_config(cfg: toml::Value) -> Fallible<Box<dyn PluginSettings>> {
+        let plugin: Self = cfg.try_into()?;
+
+        ensure!(!plugin.key_prefix.is_empty(), "empty arch-key prefix");
+        ensure!(!plugin.key_suffix.is_empty(), "empty arch-key suffix");
+
+        Ok(Box::new(plugin))
+    }
+}
+
+/// Regex for arch label validation.
+static ARCH_VALIDATION_REGEX_STR: &str = r"^[0-9a-z]+$";
+
+lazy_static! {
+    static ref ARCH_VALIDATION_REGEX_RE: regex::Regex =
+        regex::Regex::new(&ARCH_VALIDATION_REGEX_STR).expect("could not create regex");
+}
+
+impl InternalPlugin for ArchFilterPlugin {
+    fn run_internal(self: &Self, internal_io: InternalIO) -> AsyncIO<InternalIO> {
+        let future_result = futures::future::ok((
+            internal_io,
+            self.key_prefix.to_owned(),
+            self.key_suffix.to_owned(),
+        ))
+        .and_then(|(internal_io, key_prefix, key_suffix)| {
+            let arch = match try_get_multiple_values!(internal_io.parameters, "arch", "version") {
+                (Some(arch), _) => arch.to_string(),
+                (None, Some(version)) => infer_architecture_by_version(version),
+                (None, None) => {
+                    return Err(GraphError::MissingParams(
+                        vec!["arch", "version"]
+                            .into_iter()
+                            .map(String::from)
+                            .collect(),
+                    ))
+                }
+            };
+
+            if !ARCH_VALIDATION_REGEX_RE.is_match(&arch) {
+                return Err(GraphError::InvalidParams(format!(
+                    "arch '{}' does not match regex '{}'",
+                    arch, ARCH_VALIDATION_REGEX_STR
+                )));
+            };
+
+            let mut graph = internal_io.graph;
+
+            // find all releases with a different the arch
```

Comment:
```suggestion
            // find similar releases with different arch suffixes and rewrite version to create a single multiarch release
```
Not entirely sure this function does that though